contracts/modules/capital/SwapOperator.sol

```diff
@@ -347,4 +361,162 @@ contract CowSwapOperator {
+
+      // swap frequency check
+      uint timeSinceLastTrade = block.timestamp - uint(swapDetails.lastSwapTime);
+      require(timeSinceLastTrade > MIN_TIME_BETWEEN_ORDERS, "SwapOp: too fast");
+
+      uint netShareValue;
+      {
+        address denominationAsset;
+        (denominationAsset, netShareValue) =
+        enzymeFundValueCalculatorRouter.calcNetShareValue(enzymeV4VaultProxyAddress);
+
+        require(denominationAsset ==  address(weth), "SwapOp: invalid denomination asset");
+      }
+
+      // avgAmountOut in ETH
+      uint avgAmountOut = amountIn * netShareValue / (10 ** fromToken.decimals());
+      uint maxSlippageAmount = avgAmountOut * swapDetails.maxSlippageRatio / 1e18;
+      uint minOutOnMaxSlippage = avgAmountOut - maxSlippageAmount;
+
+      // slippage check
+      require(amountOutMin >= minOutOnMaxSlippage, "SwapOp: amountOutMin < minOutOnMaxSlippage");
+      require(balanceBefore > swapDetails.maxAmount, "SwapOp: balanceBefore <= max");
+      require(balanceBefore - amountIn >= swapDetails.minAmount, "SwapOp: tokenBalanceAfter < min");
+    }
+
+    pool.transferAssetToSwapOperator(address(fromToken), amountIn);
+
+    IEnzymeV4Comptroller comptrollerProxy = IEnzymeV4Comptroller(IEnzymeV4Vault(enzymeV4VaultProxyAddress).getAccessor());
+    fromToken.approve(address(comptrollerProxy), amountIn);
+
+    address[] memory payoutAssets = new address[](1);
+    uint[] memory payoutAssetsPercentages = new uint[](1);
+
+    payoutAssets[0] = address(weth);
+    payoutAssetsPercentages[0] = 10000;
+
+    comptrollerProxy.redeemSharesForSpecificAssets(address(this), amountIn, payoutAssets, payoutAssetsPercentages);
+
+    uint amountOut = weth.balanceOf(address(this));
+    weth.withdraw(amountOut);
+
+    pool.setSwapDetailsLastSwapTime(address(fromToken), uint32(block.timestamp));
+
+    require(amountOut >= amountOutMin, "SwapOp: amountOut < amountOutMin");
+
+    transferAssetTo(ETH, address(pool), amountOut);
+
+    emit Swapped(enzymeV4VaultProxyAddress, ETH, amountIn, amountOut);
+  }
+
+  function transferAssetTo (address asset, address to, uint amount) internal {
+
+    if (asset == ETH) {
+      (bool ok, /* data */) = to.call{ value: amount }("");
+      require(ok, "SwapOp: Eth transfer failed");
+      return;
+    }
+
+    IERC20 token = IERC20(asset);
+    token.safeTransfer(to, amount);
+  }
+
+  function recoverAsset(address assetAddress, address receiver) public onlyController {
+
+    IERC20 asset = IERC20(assetAddress);
+
+    uint balance = asset.balanceOf(address(this));
+    require(balance > 0, "SwapOp: Balance = 0");
+
+    IPool pool = _pool();
+
+    IPool.SwapDetails memory swapDetails = pool.getAssetSwapDetails(assetAddress);
+
+    if (swapDetails.minAmount == 0 && swapDetails.maxAmount == 0) {
+      // asset is not supported
+      asset.transfer(receiver, balance);
+      return;
+    }
```

Comment:
Would it make sense to add support for ETH as well? Could use the `transferAssetTo` function in that case.

Reply:
I would punt on this for now in the interest of time, open if you feel strongly about it

Reply:
Open a ticket for it then.